gateway/api/ray.py

```diff
@@ -174,9 +177,36 @@ def create_ray_cluster(
     namespace = settings.RAY_KUBERAY_NAMESPACE
     cluster_name = cluster_name or f"{user.username}-{str(uuid.uuid4())[:8]}"
     if not cluster_data:
+        if not job_config:
+            job_config = JobConfig()
+        if not job_config.workers:
+            job_config.workers = settings.RAY_CLUSTER_WORKER_REPLICAS
+        if not job_config.min_workers:
+            job_config.min_workers = settings.RAY_CLUSTER_WORKER_MIN_REPLICAS
+        if not job_config.max_workers:
+            job_config.max_workers = settings.RAY_CLUSTER_WORKER_MAX_REPLICAS
+        if not job_config.auto_scaling:
+            job_config.auto_scaling = settings.RAY_CLUSTER_WORKER_AUTO_SCALING
+        if not job_config.python_version:
+            job_config.python_version = settings.RAY_PYTHON_VERSION
+
+        py_version = job_config.python_version
+        node_image = (
+            settings.RAY_NODE_IMAGE[: settings.RAY_NODE_IMAGE.rindex("-")]
```

Comment:
I'm a bit worried about this. If we do not have `asdaskljdlaksjd-py39` naming convention in our images it will fail, right? 

Maybe in settings we should have a map of all available versions?
```python
# settings
...
RAY_NODE_IMAGE = os.environ.get("RAY_NODE_IMAGE", ...) # this is default version
RAY_NODE_IMAGES_MAP = {
  "PY_39": "supercool/image-py39",
  "PY_310": "...",
  ...
}
...

# api/ray.py
...
if job_config.python_version in  settings.RAY_NODE_IMAGES_MAP:
  node_image = settings.RAY_NODE_IMAGES_MAP[job_config.python_version]
else:
  logger.warn(f"Specified python version {job_config.python_version} "
    "not in a list of supported python versions {list(settings.RAY_NODE_IMAGES_MAP.keys())}. "
    "Default image will be used instead."
  )
  node_image = settings.RAY_NODE_IMAGE
...

```

What do you think?

Reply:
> I'm a bit worried about this. If we do not have asdaskljdlaksjd-py39 naming convention in our images it will fail, right?

Right.  This assumes that the image tag has "-pyxxx" postfix.  I like the idea making the map.  The image names must be configureable in the values file.   I'll work on it.  Thanks!